src/submission-service.ts

```diff
@@ -222,13 +232,42 @@ async function submit({
   autoSaveKey?: string
   onProgress?: ProgressListener
 }): Promise<FormSubmissionResult> {
+  const {
+    formSubmission,
+    shouldRunServerValidation,
+    shouldRunExternalIdGeneration,
```

Comment:
can you please move these up below `autoSaveKey` so that they are not passed on to the `submitForm()` function on line 264.